internal/database/drivers/mongodb/mongodb.go

```diff
@@ -171,28 +171,31 @@ func (m *Database) GetEvents(ctx context.Context, request requests.MultipleEvent
 	var numberOfDocuments int64
 	for _, collection := range collections {
 		var events []drivers.EiffelEvent
+		limit := request.PageSize - int32(len(allEvents))
 		findOptions := options.Find().
 			SetProjection(bson.M{"_id": 0}).
 			SetSkip(int64((request.PageNo - 1) * request.PageSize)).
-			SetLimit(int64(request.PageSize))
+			SetLimit(int64(limit))
```

Comment:
The limit could get negative. Are we sure that won't upset SetLimit? Moving the definition of findOptions inside the conditional on line 181 would address that and perhaps also make the code more readable.

Reply:
Won't upset SetLimit, but readability is better as you said.